standard-covoiturage_openapi.yaml

```diff
@@ -116,6 +128,301 @@ paths:
+                  allOf:
+                    - $ref: '#/components/schemas/PassengerTrip'
+                    - type: object
+                      properties:
+                        schedules:
+                          type: array
+                          items:
+                            $ref: '#/components/schemas/Schedule'
+        400:
+          $ref: '#/components/responses/BadRequest'
+        401:
+          $ref: '#/components/responses/Unauthorized'
+        429:
+          $ref: '#/components/responses/TooManyRequests'
+        500:
+          $ref: '#/components/responses/InternalServerError'
+
+  /messages:
+    post:
+      tags:
+      - Interact
+      summary: Send a message to the owner of a retrieved journey.
+      description: Route used to allow a user to connect back to the owner of a retrieved journey through a texte message.
+      operationId: postConnections
+      requestBody:
+        content:
+          application/json:
+            schema:
+              allOf:
+                - $ref: '#/components/schemas/ConnectionMessage'
+              type: object
+              required:
+                - message
+                - recipientCarpoolerType
+              properties:
+                recipientCarpoolerType:
+                  type: string
+                  enum:
+                    - driver
+                    - passenger
+                  default: driver
+                  description: Defines if the recipient of this message is either the driver or the passenger.
+      responses:
+        201:
+          description: Successful operation.
+        404:
+          description: The targeted journey or user no more exists.
+        400:
+          $ref: '#/components/responses/BadRequest'
+        401:
+          $ref: '#/components/responses/Unauthorized'
+        429:
+          $ref: '#/components/responses/TooManyRequests'
+        500:
+          $ref: '#/components/responses/InternalServerError'
+
+  /bookings:
+    post:
+      tags:
+      - Interact
+      summary: Create a punctual outward Booking request.
+      description: Route used to synchronize a Booking request initiated by a platform to the second platform involved in the shared punctual outward journey. While posting a new Booking, its status shall always be set first as `status=WAITING_CONFIRMATION`. The  Booking MUST be created using the relevant `passengerJourneyId` or `driverJourneyId` attributes according to the operator role handling the request (the operator can recognize its own role thanks to the `driverOperator` and `passengerOperator` attributes).
+        _Reminder:_ In case of booking without deeplink, the sender platform MUST also store the Booking object, and be ready to receive modifications of it through the PATCH /bookings API endpoint.
+      operationId: postBookings
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Booking'
```

Comment:
I wonder if it would not simplify if we create a `BookingRequest` (as the description tells : Create a punctual outward Booking request). From my point of view, journey ID and User ID should be sufficient here. 

In this case, only the MSP would be responsible of creating a BookingID, and I don't see the point of copying all journey information into the request. What should happen if the information is not the same as the one associated with the journey Id for instance ?

Reply:
As of the discussion of today, we decided to keep a Booking-like object, where the journey ID is only informative and the information inside the booking is authoritative. This allows more flexibility, and offers support to the case where there has been no search beforehand - cf [this comment](https://github.com/fabmob/standard-covoiturage/pull/27#issuecomment-1128796283)